Bluetooth: shell: improve a2dp shell and readme

MarkWangChinese opened the pull request:
Before every cmd, call one common function to check whether a2dp is initialized.
Add abort req and rsp callbacks, improve register_ep cmd to support delay report, check buf tailroom for sending media, add get_conn cmd to test bt_a2dp_get_conn.
Improve a2dp shell doc

MarkWangChinese committed:
bluetooth: shell: a2dp: use common function to check a2dp initialization

Before every cmd, call one common function to check whether a2dp is
initialized.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

MarkWangChinese committed:
bluetooth: shell: a2dp: improve a2dp shell cmds

add abort req and rsp callbacks, improve register_ep cmd to support delay
report, check buf tailroom for sending media, add get_conn cmd to test
bt_a2dp_get_conn.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

MarkWangChinese committed:
doc: Bluetooth: shell: improve a2dp shell doc

describe all the a2dp shell cmds.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

MarkWangChinese committed:
Bluetooth: shell: a2dp: Update a2dp sdp profile version

The newest a2dp spec set the profile version as 1.4.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -874,13 +916,32 @@ static int cmd_get_config(const struct shell *sh, int32_t argc, char *argv[])
+		shell_print(sh, "a2dp conn is: %p", conn);
+	} else {
+		shell_print(sh, "a2dp conn is: NULL");
+	}
```
`bt_conn_unref` is required.

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -118,7 +118,7 @@ static struct bt_sdp_attribute a2dp_sink_attrs[] = {
 			{
 				BT_SDP_TYPE_SIZE(BT_SDP_UINT16), /* 09 */
-				BT_SDP_ARRAY_16(0x0103U) /* 01 03 */
+				BT_SDP_ARRAY_16(0x0104U) /* 01 04 */
```
If a macro is more clear?

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -30,7 +30,7 @@
 static uint8_t a2dp_sink_sdp_registered;
 static uint8_t a2dp_source_sdp_registered;
-static uint8_t a2dp_initied;
+static bool a2dp_initied;
```
If is it `a2dp_initiated`?

lylezhu2012 replied:
If the variable name `a2dp_cb_registered` is more reasonable?

makeshi commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -499,6 +501,22 @@ static void app_get_config_rsp(struct bt_a2dp_stream *stream, struct bt_a2dp_cod
+	if (rsp_err_code == 0) {
+		bt_shell_print("success to abort");
+	} else {
+		bt_shell_print("fail to abort");
```
shell_error()?

MarkWangChinese committed:
bluetooth: shell: a2dp: use bt_shell_xxx instead of shell_xxx

Replace all shell_print() and shell_error() calls with bt_shell_print()
and bt_shell_error() respectively to use the common Bluetooth shell
printing functions throughout the A2DP shell implementation.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

MarkWangChinese committed:
bluetooth: shell: a2dp: use bt_shell_error for error messages

Replace bt_shell_print() calls with bt_shell_error() for all error
conditions and failure cases in the A2DP shell implementation to
properly distinguish error messages from informational output.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -499,6 +501,22 @@ static void app_get_config_rsp(struct bt_a2dp_stream *stream, struct bt_a2dp_cod
+		bt_shell_print("success to abort");
+	} else {
+		bt_shell_error("fail to abort");
+	}
```
I suggest the error code `rsp_err_code` could be printed if it is not zero.

MarkWangChinese committed:
bluetooth: shell: a2dp: include error code in failure messages

Add error code parameter to all A2DP shell error messages to provide
more detailed diagnostic information when operations fail. This applies
to configure, establish, release, start, suspend, delay report, and
abort response handlers.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

MarkWangChinese commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -187,7 +189,7 @@ static struct bt_sdp_attribute a2dp_source_attrs[] = {
 			{
 				BT_SDP_TYPE_SIZE(BT_SDP_UINT16),
-				BT_SDP_ARRAY_16(0x0103U)
+				BT_SDP_ARRAY_16(0x0104U)
```
Done

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -546,7 +546,7 @@ static struct bt_a2dp_cb a2dp_cb = {
 {
 	if (!a2dp_cb_registered) {
-		shell_print(sh, "need to register a2dp connection callbacks");
+		bt_shell_print("need to register a2dp connection callbacks");
```
```suggestion
		bt_shell_error("need to register a2dp connection callbacks");
```

MarkWangChinese replied:
Done

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -562,12 +562,12 @@ static int cmd_register_cb(const struct shell *sh, int32_t argc, char *argv[])
+			bt_shell_print("success");
 		} else {
-			shell_print(sh, "fail");
+			bt_shell_print("fail");
```
```suggestion
			bt_shell_error("fail");
```

MarkWangChinese replied:
Done

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -641,7 +641,7 @@ static int cmd_register_ep(const struct shell *sh, int32_t argc, char *argv[])
 
 	if (err) {
-		shell_print(sh, "fail to register endpoint");
+		bt_shell_print("fail to register endpoint");
```
```suggestion
		bt_shell_error("fail to register endpoint");
```

MarkWangChinese replied:
Done

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -562,12 +562,12 @@ static int cmd_register_cb(const struct shell *sh, int32_t argc, char *argv[])
 		}
 	} else {
-		shell_print(sh, "already registered");
+		bt_shell_print("already registered");
```
```suggestion
		bt_shell_error("already registered");
```

MarkWangChinese replied:
Done

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -810,7 +810,7 @@ static int cmd_release(const struct shell *sh, int32_t argc, char *argv[])
 
 	if (bt_a2dp_stream_release(&sbc_stream) != 0) {
-		shell_print(sh, "fail");
+		bt_shell_print("fail");
```
```suggestion
		bt_shell_error("fail");
```

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -834,7 +834,7 @@ static int cmd_suspend(const struct shell *sh, int32_t argc, char *argv[])
 
 	if (bt_a2dp_stream_suspend(&sbc_stream) != 0) {
-		shell_print(sh, "fail");
+		bt_shell_print("fail");
```
```suggestion
		bt_shell_error("fail");
```

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -846,7 +846,7 @@ static int cmd_abort(const struct shell *sh, int32_t argc, char *argv[])
 
 	if (bt_a2dp_stream_abort(&sbc_stream) != 0) {
-		shell_print(sh, "fail");
+		bt_shell_print("fail");
```
```suggestion
		bt_shell_error("fail");
```

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -899,7 +899,7 @@ static int cmd_send_delay_report(const struct shell *sh, int32_t argc, char *arg
 	err = bt_a2dp_stream_delay_report(&sbc_stream, 1);
 	if (err < 0) {
-		shell_print(sh, "fail to send delay report (%d)\n", err);
+		bt_shell_print("fail to send delay report (%d)\n", err);
```
```suggestion
		bt_shell_error("fail to send delay report (%d)\n", err);
```

MarkWangChinese commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -798,7 +798,7 @@ static int cmd_establish(const struct shell *sh, int32_t argc, char *argv[])
 
 	if (bt_a2dp_stream_establish(&sbc_stream) != 0) {
-		shell_print(sh, "fail");
+		bt_shell_print("fail");
```
Done

MarkWangChinese commented on subsys/bluetooth/host/classic/shell/a2dp.c:
```diff
@@ -342,7 +342,7 @@ static void app_config_rsp(struct bt_a2dp_stream *stream, uint8_t rsp_err_code)
 		bt_shell_print("success to configure");
 	} else {
-		bt_shell_error("fail to configure");
+		bt_shell_error("fail to configure (err code %d)", rsp_err_code);
```
Done
